packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -346,6 +346,32 @@ class TextSelectionOverlay {
 
   TextSelection get _selection => _value.selection;
 
+  /// Opacity of selection handles.
+  ///
+  /// Set to 0.0 if you want to hide the handles.
+  ///
+  /// If this method is called while the [SchedulerBinding.schedulerPhase] is
+  /// [SchedulerPhase.persistentCallbacks], i.e. during the build, layout, or
+  /// paint phases (see [WidgetsBinding.drawFrame]), then the update is delayed
+  /// until the post-frame callbacks phase. Otherwise the update is done
+  /// synchronously. This means that it is safe to call during builds, but also
+  /// that if you do call this during a build, the UI will not update until the
+  /// next frame (i.e. many milliseconds later).
+  double get handlesOpacity => _handlesOpacity;
+  double _handlesOpacity = 1.0;
+  set handlesOpacity(double opacity) {
+    if (_handlesOpacity == opacity)
+      return;
+    _handlesOpacity = opacity;
+    // If we are in build state, it will be too late to update opacity.
+    // We will need to schedule the build in next frame.
+    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
```

Comment:
unfortunately, i still have to do this post frame trick to support setState during handleSelectionChange.